### DISCUSSION TITLE
Manipulation of incoming request data  ( Multipart request ) using policy.

### DISCUSSION THREAD

**Discussion by iamakashk** [apiman/apiman#1319] [Help] [answered]
Hi team,

I m working on creation of custom policy issue .
So in the policy i have to get the details of multipart data , which consist of a json data and an Image file ( single or multiple).

I need to modify the incoming json for some key ( example : like attaching log id to json data ) .

To achieve this i have tried to add the chunk data to a byte array.
```
                     @Override
			public void write(IApimanBuffer chunk) {
		
				String requestType = request.getType();
				String contentType = request.getHeaders().get(CommonConstants.CONTENT_TYPE);

				if(requestType.equals(CommonConstants.POST_METHOD)) {
					byte[] bytes = chunk.getBytes();
					readBuffer.append(bytes);
				}else {
					super.write(chunk);
				}
			}
```

and in  "end" Method i am fetching the **readBuffer** and able to get json data . 
Now the problem is I unable to put all the read buffer data all together and send it back to backend .

So is there anyway to read the chunk data from apiman , and modify it without disturbing the multipart format.

**Comment by msavy** [comment, marked as answer]
Have you checked out the TransformationPolicy plugin as an example that might be helpful to you?

**Reply by msavy** [reply]
Did you resolve this @iamakashk ?

**Reply by iamakashk** [reply]
@msavy  Could not able to do this. Still working on it. Basically able to manipulate json but file is somehow not getting forwarded corrected when I combine everything again. Will update you on this.
